packages/camera/camera_platform_interface/test/types/resolution_preset_test.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:camera_platform_interface/camera_platform_interface.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('ResolutionPreset should contain 6 options', () {
+    final values = ResolutionPreset.values;
+
+    expect(values.length, 6);
+  });
+
+  test("ResolutionPreset enum should have items in correct index", () {
```

Comment:
Why? Is the index passed around through the method channel? (I'd recommend to *not* do that, and instead pass a String version of the enum value, if possible!)

Reply:
Resolution presets are communicated as strings so that is not really an issue. I am just used to make sure not to change an index of an enum since it will result in a breaking change of the API (theoretically developers could also access the index of the enum and use it in al kinds of ways), so I like to have a unit-test to guard against index changes.

Reply:
Wow, I hand't thought of this... I sure have a bunch of "untested" enums around! :P